### PR TITLE
[Docs]zh-cn document translate: squeezing avr page and keycodes basic page

### PR DESCRIPTION
**Pull request by givedrug**
## Description

zh-cn document translate: squeezing avr page

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* tracking issue:  #15443

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
~~- [ ] My change requires a change to the documentation.~~
~~- [ ] I have updated the documentation accordingly.~~
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
~~- [ ] I have added tests to cover my changes.~~
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/zh-cn/squeezing_avr.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+#define NO_MUSIC_MODE
+```
+同时修改 `rules.mk`：
+```make
+MUSIC_ENABLE = no
+```
+
+## Layers层配置
+
+*层*也有很多配置来减少固件空间，所有配置在`config.h`。
```
Comment (peepeetee) [inline docs/zh-cn/squeezing_avr.md:73]:
```suggestion
*层*也有很多配置来减少固件空间，所有配置在`config.h`中。
```

File: docs/zh-cn/squeezing_avr.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
+# 充分利用AVR
+
+AVR对空间限制极其严格，随着QMK的持续开发，无法再适应这些限制，所以对AVR的支持可能会成为历史。
```
Comment (peepeetee) [inline docs/zh-cn/squeezing_avr.md:3]: reword this somehow

读起来不太对劲，重新试试

File: docs/zh-cn/squeezing_avr.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+```c
+    // NEW CODE
+    oled_write_P(PSTR("WPM: "), false);
+    oled_write(get_u8_str(get_current_wpm(), '0'), false);
+```
+将会输出 `WPM: 005`.
+
+## RGB配置
+
+如果你的板子使用了GRB，无论是RGB灯（底灯）还是RGB矩阵（按键灯）都需要很多配置去定义不同的动效。有一些键盘默认允许大量的动效，所以可以通过禁用你不需要的动效来节省点空间。
```
Comment (peepeetee) [inline docs/zh-cn/squeezing_avr.md:123]:
```suggestion
如果你的板子使用了RGB，无论是RGB灯（底灯）还是RGB矩阵（按键灯）都需要很多配置去定义不同的动效。有一些键盘默认允许大量的动效，所以可以通过禁用你不需要的动效来节省点空间。
```

**Review by peepeetee — CHANGES_REQUESTED** [review]

File: docs/zh-cn/squeezing_avr.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
+# 充分利用AVR
+
```
Comment (peepeetee) [inline docs/zh-cn/squeezing_avr.md:2]:
```suggestion

<!---
  original document: 0.14.23:docs/keycodes.md
  git diff 0.14.23 HEAD -- docs/keycodes.md | cat
-->

```

**Review by peepeetee — CHANGES_REQUESTED** [review]

File: docs/zh-cn/squeezing_avr.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+  original document: 0.15.12:docs/squeezing_avr.md
+  git diff 0.15.12 HEAD -- docs/squeezing_avr.md | cat
+-->
+
+AVR对空间限制极其严格，随着QMK的持续开发，无法再适应这些限制，所以对AVR的支持可能会成为历史。
+
+话说回来，如果想要减小固件编译后的大小，你可以采取如下操作：
+
+## `rules.mk` 配置
+第一步，也是非常关键的一步，允许*连接时间优化*(link time optimizatio)，在rules.mk中添加如下配置：
```
Comment (epiciskandar) [inline docs/zh-cn/squeezing_avr.md:13]: 建议：链接时优化

File: docs/zh-cn/squeezing_avr.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+```make
+LTO_ENABLE = yes
+```
+打开此配置后，会让最后的编译操作花费更长的时间，但是生成的编译文件会更小。同时还会禁用*活动方法*(Action Functions)和*活动宏*(Action Macros)，两者都已不建议使用。
```
Comment (epiciskandar) [inline docs/zh-cn/squeezing_avr.md:18]: 建议：Function通常翻译为函数

File: docs/zh-cn/squeezing_avr.md (hunk 5 above)
Comment (givedrug) [reply docs/zh-cn/squeezing_avr.md:13]: accept

File: docs/zh-cn/squeezing_avr.md (hunk 6 above)
Comment (givedrug) [reply docs/zh-cn/squeezing_avr.md:18]: accept

File: docs/zh-cn/keycodes_basic.md [hunk 7]
```diff
@@ -0,0 +1,256 @@
+# 基础键码
+
```
Comment (daskygit) [inline docs/zh-cn/keycodes_basic.md:2]:
```suggestion

<!---
  original document: 0.15.18:docs/keycodes_basic.md
  git diff 0.15.18 HEAD -- docs/keycodes_basic.md | cat
-->

```

File: docs/zh-cn/keycodes_basic.md [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+| `KC_SEPARATOR`       | `KC_SEPR`                      | Separator键                               |
+| `KC_OUT`             |                                | Out键                                     |
+| `KC_OPER`            |                                | Oper键                                    |
+| `KC_CLEAR_AGAIN`     | `KC_CLAG`                      | Clear/Again键                             |
+| `KC_CRSEL`           | `KC_CRSL`                      | CrSel/Props键                             |
+| `KC_EXSEL`           | `KC_EXSL`                      | ExSel键                                   |
+
+## 媒体键
+
+这些键码不在**键盘/小型键盘使用手册**的*使用（Usage）*章节中。`SYSTEM_` 前缀的键码在*通用台式机（Generic Desktop）*章节，其余部分在*用户（Consumer）*章节。
```
Comment (daskygit) [inline docs/zh-cn/keycodes_basic.md:192]: I'm unsure on this translation, I'd like to see some input from others if this is okay.

File: docs/zh-cn/squeezing_avr.md [hunk 9]
```diff
@@ -0,0 +1,209 @@
+# 充分利用AVR
+
+<!---
+  original document: 0.15.12:docs/squeezing_avr.md
+  git diff 0.15.12 HEAD -- docs/squeezing_avr.md | cat
```
Comment (daskygit) [inline docs/zh-cn/squeezing_avr.md:5]:
```suggestion
  original document: 0.15.18:docs/squeezing_avr.md
  git diff 0.15.18 HEAD -- docs/squeezing_avr.md | cat
```

**Review by daskygit — CHANGES_REQUESTED** [review]

File: docs/zh-cn/keycodes_basic.md [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+| `KC_F23` | F23  |
+| `KC_F24` | F24  |
+
+## 符号键
+
+|键                 |别名  | 描述                      |
+|--------------------|---------|-------------------------|
+|`KC_ENTER`          |`KC_ENT` | 回车键（Return 、Enter）    |
+|`KC_ESCAPE`         |`KC_ESC` | 退出键（Escape）             |
+|`KC_BACKSPACE`      |`KC_BSPC`| 删除键或退格键（Delete 、Backspace） |
```
Comment (peepeetee) [inline docs/zh-cn/keycodes_basic.md:81]:
```suggestion
|`KC_BACKSPACE`      |`KC_BSPC`| 删除键或退格键（Backspace） |
```

**Review by peepeetee — CHANGES_REQUESTED** [review]

File: docs/zh-cn/keycodes_basic.md (hunk 8 above)
Comment (epiciskandar) [inline docs/zh-cn/keycodes_basic.md:192]: `Keyboard/Keypad page` is a unique paragraph title, IMO we should not split them apart here. but the embarrassing situration is that the word `usage` here translated to `使用`(which more like verb `use`) is not quite right, but we have no percedent or example to following.
sentences after that looks OK to me.

Comment (epiciskandar) [reply docs/zh-cn/keycodes_basic.md:192]: my suggestion:

```
这些键码不在*键盘/小型键盘（Keyboard/Keypad page）*章节中。
```

i.e. keep originals, skip the `usage`, it's not quite important in Chinese here.

**Comment by epiciskandar** [comment]
keep this

**Pull request closed without merging**